Dockerfile.updater-core

```diff
@@ -10,12 +10,25 @@ ENV DEBIAN_FRONTEND="noninteractive" \
   LC_ALL="en_US.UTF-8" \
   LANG="en_US.UTF-8"
 
+# Use a custom sources.list to use Azure's mirrors which are faster
+ADD ubuntu.sources /etc/apt/sources.list.d/ubuntu.sources
+
 RUN apt-get update \
   && apt-get upgrade -y \
   && apt-get install -y --no-install-recommends \
+    # needed to run add-apt-repository
+    software-properties-common \
+    # Used temporarily to download git lfs install script as well as dev dependencies for CI
+    curl \
+    # Add git core ppa to get a more recent git version than the one provided by ubuntu 24.04 which is 2.43.0(released in November 20, 2023)
+    && add-apt-repository -y ppa:git-core/ppa \
+    && apt-get update \
+    # Download the installation script for git-lfs and install it. See https://github.com/git-lfs/git-lfs/blob/main/INSTALLING.md \
+    # We need this because the version of git-lfs provided by ubuntu 24.04 outdated(3.4.1, released in Dec 13, 2023)
+    && curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash \
```

Comment:
@JamieMagee I understand that the main objective was to upgrade Git, but I thought it would make sense to update `git-lfs` at the same time to prevent potential issues. If you’d prefer, I can handle that upgrade separately

Reply:
I share your concern, especially since it involves arbitrary external code.

I found this reference in the official documentation: https://github.com/git-lfs/git-lfs/blob/main/INSTALLING.md

There may be other ways to install it, but at least that's what it is directly documented. 

One approach to make it safer could be to extract only the parts we need from their script, even if it requires additional steps.  The downside is that it would increase our maintenance overhead to keep that up to date

Reply:
Yeah, looking at the script the key parts are:
1. Getting the gpg key from [here][1] and storing it in `/etc/ap/keyrings/github_git-lfs-archive-keyring.gpg`
2. Creating `/etc/apt/sources.list.d/github_git-lfs.list` with the content

    ```
    deb [signed-by=/etc/apt/keyrings/github_git-lfs-archive-keyring.gpg] https://packagecloud.io/github/git-lfs/ubuntu/ noble main
    deb-src [signed-by=/etc/apt/keyrings/github_git-lfs-archive-keyring.gpg] https://packagecloud.io/github/git-lfs/ubuntu/ noble main
    ```

[1]: https://packagecloud.io/github/git-lfs/gpgkey